core/src/test/java/org/elasticsearch/search/aggregations/InternalMultiBucketAggregationTestCase.java

```diff
@@ -57,7 +73,13 @@ public void setUp() throws Exception {
 
     @Override
     protected final T createTestInstance(String name, List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) {
-        return createTestInstance(name, pipelineAggregators, metaData, subAggregationsSupplier.get());
+        final int maxNumberOfBuckets = maxNumberOfBuckets();
+        T instance = createTestInstance(name, pipelineAggregators, metaData, subAggregationsSupplier.get());
+
+        int numberOfBuckets = instance.getBuckets().size();
+        assertThat("Expecting a maximum number of "+ maxNumberOfBuckets + " buckets for " + instance.getClass().getSimpleName()
+                + " aggregation but got " + numberOfBuckets, numberOfBuckets, lessThanOrEqualTo(maxNumberOfBuckets));
```

Comment:
Can we make this a regular "assert" since it checks some assumptions about correct test setup rather than something in the aggregation we are testing.